components/server/src/auth/auth-provider-service.ts

```diff
@@ -56,73 +60,268 @@ export class AuthProviderService {
         const result = await this.authProviderDB.findByOrgId(organizationId);
         return result;
     }
 
-    async deleteAuthProvider(authProvider: AuthProviderEntry): Promise<void> {
+    async deleteAuthProviderOfUser(userId: string, authProviderId: string): Promise<void> {
+        await this.auth.checkPermissionOnUser(userId, "write_info", userId);
+
+        const ownProviders = await this.getAuthProvidersOfUser(userId);
+        const authProvider = ownProviders.find((p) => p.id === authProviderId);
+        if (!authProvider) {
+            throw new ApplicationError(ErrorCodes.NOT_FOUND, "User resource not found.");
+        }
+
         await this.authProviderDB.delete(authProvider);
     }
 
-    async updateAuthProvider(
-        entry: AuthProviderEntry.UpdateEntry | AuthProviderEntry.NewEntry,
-    ): Promise<AuthProviderEntry> {
-        let authProvider: AuthProviderEntry;
-        if ("id" in entry) {
-            const { id, ownerId } = entry;
-            const existing = (await this.authProviderDB.findByUserId(ownerId)).find((p) => p.id === id);
-            if (!existing) {
-                throw new Error("Provider does not exist.");
-            }
-            const changed =
-                entry.clientId !== existing.oauth.clientId ||
-                (entry.clientSecret && entry.clientSecret !== existing.oauth.clientSecret);
+    async deleteAuthProviderOfOrg(userId: string, organizationId: string, authProviderId: string): Promise<void> {
+        await this.auth.checkPermissionOnOrganization(userId, "write_git_provider", organizationId);
 
-            if (!changed) {
-                return existing;
-            }
+        // Find the matching auth provider we're attempting to delete
+        const orgProviders = await this.getAuthProvidersOfOrg(userId, organizationId);
+        const authProvider = orgProviders.find((p) => p.id === authProviderId && p.organizationId === organizationId);
+        if (!authProvider) {
+            throw new ApplicationError(ErrorCodes.NOT_FOUND, "Provider resource not found.");
+        }
 
-            // update config on demand
-            const oauth = {
-                ...existing.oauth,
-                clientId: entry.clientId,
-                clientSecret: entry.clientSecret || existing.oauth.clientSecret, // FE may send empty ("") if not changed
-            };
-            authProvider = {
-                ...existing,
-                oauth,
-                status: "pending",
-            };
+        await this.authProviderDB.delete(authProvider);
+    }
+
+    async getAuthProvider(userId: string, id: string): Promise<AuthProviderEntry | undefined> {
+        const result = await this.authProviderDB.findById(id);
+        if (!result) {
+            return undefined;
+        }
+
+        if (result.organizationId) {
+            await this.auth.checkPermissionOnOrganization(userId, "write_git_provider", result.organizationId);
```

Comment:
should it be here read_git_provider and red_info?

Reply:
good catch!